### PR TITLE
feat: add triggerFormValidation() API for dynamic form structure changes

### PR DESCRIPTION
**Pull request by the-ult**
### Problem Solved

Previously, when form structure changed dynamically (e.g., conditional fields shown/hidden) without control value changes, validation state wouldn't update automatically. This occurred because Angular's `formValueChange` event isn't emitted when only DOM structure changes.

**Example scenario**: Switching from a form with required input fields to one with only informational content (no inputs) would leave the form in an invalid state, even though no additional validation was needed.

#### ✨ New Public Method

```typescript
public triggerFormValidation(): void
```

Manually triggers form validation update when form structure changes but no control values change.

#### 🔧 Usage Pattern

```typescript
@Component({
  template: `
    <form scVestForm #vestForm="scVestForm" [suite]="suite">
      <select (ngModelChange)="onProcedureTypeChange($event)">
        <option value="typeA">Type A (requires input)</option>
        <option value="typeC">Type C (informational only)</option>
      </select>
      
      @if (formValue().procedureType === 'typeA') {
        <input name="fieldA" [ngModel]="formValue().fieldA" />
      } @else if (formValue().procedureType === 'typeC') {
        <p>No additional input required.</p>
      }
    </form>
  `
})
export class MyComponent {
  @ViewChild('vestForm') vestForm!: FormDirective<MyFormModel>;

  onProcedureTypeChange(newType: string) {
    this.formValue.update(current => ({ ...current, procedureType: newType }));
    // ✅ CRITICAL: Trigger validation after structure change
    this.vestForm.triggerFormValidation();
  }
}
```

#### Core Implementation
- **`FormDirective`**: Added `triggerFormValidation()` public method with comprehensive JSDoc
- **`FormDirective`**: Added `exportAs: 'scVestForm'` for template reference access
- **`FormDirective`**: Enhanced formValueChange with better change detection logic

#### Comprehensive Documentation
- **`README.md`**: New "Handling Form Structure Changes" section with examples
- **`README.md`**: Added Public Methods API table
- **`README.md`**: Enhanced Documentation section with organized guide links
- **`docs/STRUCTURE_CHANGE_DETECTION.md`**: Advanced guide with alternatives and best practices

#### Testing & Examples
- **Storybook Stories**: Complete demonstration of issue and solution patterns
- **Unit Tests**: Coverage for new functionality and edge cases
- **Angular 20 Best Practices**: Examples using signals, `@if`, `ChangeDetectionStrategy.OnPush`

### 🎯 Use Cases

Perfect for:
- ✅ Conditionally showing/hiding form controls
- ✅ Adding/removing form controls dynamically  
- ✅ Switching between different form layouts
- ✅ Clearing form sections programmatically
- ✅ Multi-step forms with varying validation requirements

### 🔄 When to Use

```typescript
// Call after any of these scenarios:
this.vestForm.triggerFormValidation();
```

1. **After changing form structure** - Conditional fields shown/hidden
2. **After clearing form sections** - Resetting parts of the form
3. **After dynamic field addition/removal** - Programmatic form changes
4. **After switching form modes** - Different form layouts/requirements

### 🚀 Performance

- **Zero overhead** when not called
- **Explicit control** - only triggers when needed
- **Efficient** - much better than polling-based solutions
- **Follows Angular patterns** - works with existing form validation lifecycle

### 📚 Documentation Strategy

- **README.md**: Basic usage and essential patterns
- **docs/**: Advanced guides and detailed alternatives
- **Storybook**: Interactive examples and edge cases
- **Unit Tests**: Comprehensive coverage and usage patterns

### 🔖 Version Impact

This introduces a new public API method, making it a **minor version bump** to `1.3.0`:
- ✅ Backward compatible
- ✅ New functionality
- ✅ Enhanced developer experience
- ✅ Solves real-world form validation challenges

### 🧪 Testing

- **Unit tests** for new functionality
- **Storybook interaction tests** for UI validation
- **Edge case coverage** for all structure change scenarios
- **Angular 20 compatibility** verified

---

**Breaking Changes**: None ✅  
**Migration Required**: None ✅  
**New Dependencies**: None ✅

### CODE REVIEW

File: projects/ngx-vest-forms/src/lib/directives/form.directive.ts [hunk 1]
```diff
@@ -210,6 +214,39 @@ export class FormDirective<T extends Record<string, any>>
...
+   * onProcedureTypeChange(newType: string) {
+   *   this.procedureType.set(newType);
+   *   /// Structure changed but no control values changed
+   *   this.formDirective.triggerFormValidation();
+   * }
+   * ```
+   */
+  public triggerFormValidation(): void {
+    // Update all form controls validity which will trigger all form events
+    this.ngForm.form.updateValueAndValidity({ emitEvent: true });
```
Comment (JoeriRoijenga) [inline projects/ngx-vest-forms/src/lib/directives/form.directive.ts:247]: Won't this trigger the validation for all form fields? Is the desirable?

Comment (the-ult) [reply projects/ngx-vest-forms/src/lib/directives/form.directive.ts:278]: Angular Forms Architecture
Per Angular's documentation and best practices, updateValueAndValidity() on the form root is the recommended approach for structural changes because:

It ensures all validators are re-evaluated with the current form state
It triggers all necessary form events (StatusChangeEvent, etc.)
It maintains form consistency and prevents stale validation states
